PurchasesCoreSwift/Public/PurchaserInfo.swift

```diff
@@ -0,0 +1,326 @@
+    Use this for grandfathering users when migrating to subscriptions.
+
+     
+     @note This can be nil, see -[RCPurchases restoreTransactionsForAppStore:]
+     */
+    @objc public let originalApplicationVersion: String?
+
+    // TODO after migration make this internal
+    @objc public let schemaVersion: String?
+
+    private let expirationDatesByProduct: [String: Date?]
+
+    private let purchaseDatesByProduct: [String: Date?]
+
+    private let originalData: [String: Any]
+
+    private static var dateFormatter: DateFormatter {
+        let formatter = DateFormatter()
+        formatter.timeZone = TimeZone.init(secondsFromGMT: 0)
+        formatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ssZ"
+        formatter.locale = Locale(identifier: "en_US_POSIX")
+        return formatter
+    }
+
+    @objc public init?(data: [String: Any]) {
+        guard let subscriberObject = data["subscriber"] as? [String: Any],
+              let subscriberData = SubscriberData.init(subscriberData: subscriberObject)
+            else {
+            return nil
+        }
+
+        self.originalData = data
+        self.schemaVersion = data["schema_version"] as? String
+
+        guard let requestDateString = data["request_date"] as? String,
+              let formattedRequestDate = Self.dateFormatter.date(from: requestDateString) else {
+            return nil
+        }
+        self.requestDate = formattedRequestDate
+
+        self.originalPurchaseDate = subscriberData.originalPurchaseDate
+        self.firstSeen = subscriberData.firstSeen
+        self.originalAppUserId = subscriberData.originalAppUserId
+        self.managementURL = subscriberData.managementURL
+        self.originalApplicationVersion = subscriberData.originalApplicationVersion
+
+        self.nonConsumablePurchases = Set(subscriberData.nonSubscriptions.keys)
+        self.nonSubscriptionTransactions = subscriberData.nonSubscriptionTransactions
+
+        self.entitlements = EntitlementInfos(entitlementsData: subscriberData.entitlements,
+                                             purchasesData: subscriberData.allPurchases,
+                                             dateFormatter: Self.dateFormatter,
+                                             requestDate: requestDate)
+
+        self.expirationDatesByProduct = subscriberData.expirationDatesByProduct
+        self.purchaseDatesByProduct = subscriberData.purchaseDatesByProduct
+    }
+
+    // TODO after migration make this internal
+    @objc public class func currentSchemaVersion() -> String { "2" }
+
+    /**
+     Get the expiration date for a given product identifier. You should use Entitlements though!
+    
+     @param forProductIdentifier Product identifier for product
+    
+     @return The expiration date for `productIdentifier`, `nil` if product never purchased
+     */
+    @objc public func expirationDate(forProductIdentifier: String) -> Date? {
```

Comment:
`forProductIdentifier` doesn't really work well as a parameter name. If we want to keep this naming for `objc` compatibility, then we should consider addding a label, i.e.: 

```swift
@objc public func expirationDate(forProductIdentifier productIdentifier: String) -> Date? {
```

Reply:
same thinking for other methods that have `forFoo` as the param name